### ISSUE TITLE
Setting  --user-data-dir to custom location still causes  AppData folder to be created

### ISSUE THREAD

**Issue by starsoccer** [nwjs/nw.js#6328]
NWJS Version : 0.27.1
Operating System : Windows 10

### Expected behavior

If user-data-dir is set to a custom location, no folder should be created in AppData\Local\%appname%

### Actual behavior

Empty folders are created in AppData\Local\%appname%. Specifically a User Data and a User Data\CrashPad folder are created

### How to reproduce

1. Create a simple app with NWJS and then set the  --user-data-dir to a custom location.
2. Run the app
3. Close the app
4. Check AppData\Local\%appname% to see if a folder exists

**Comment by Christywl** [comment]
I can reproduce this issue on Linux/Windows with nwjs-sdk-v0.27.1.

**Comment by rowenbuzz** [comment]
I would like to add a [Feature Request] to simply not write the app_name (where app_name is the name field value in the package.json) not only from the \\Local\AppData location, but do not write it anywhere if possible.

Here is a thread on the Google Group explaining more details with Roger:
https://groups.google.com/forum/#!topic/nwjs-general/bQdFZEVXcBs

**Comment by zkrige** [comment]
> Is this bug fixed yet? nw.js version 0.31.4 still has this problem, that's annoying.
> By the way, `nw.exe --user-data-dir=%APPDATA%\myapp` works well with `cmd.exe`, but `"chromium-args":"--user-data-dir=%APPDATA%\\myapp"` in the `package.json` didnt work.
> It just creates a folder named %APPDATA% in the current directory but not converts to system path, is that a bug or I'm using it in a wrong way?

I'm having the same issue. The environment variables aren't being expanded, but are being used as literals. nwjs:0.32.4

**Comment by mikekilburn** [comment]
@rogerwang is there a plan to fix this? Or at least offer an nwjs api/setting to use %APPDATA% rather than %LOCALAPPDATA% which is the default now. This bug makes it impossible to have the user-data-dir in each users roaming folder.

**Comment by f3l1x** [comment]
Version: `0.45.5`

```json
{
  "name": "test",
  "version": "0.0.1",
  "main": "app.js"
  "chromium-args": [
    "--user-data-dir=./data"
  ]
}
```

Does not work for me.

**Comment by mikekilburn** [comment]
@rogerwang here is a pull request that fixes this issue. https://github.com/nwjs/nw.js/pull/7477 . At Mitel we use nwjs for our UC client and need to have local storage in the roaming folder (%APPDATA%) rather than %LOCALAPPDATA%. Please have a look.

**Comment by TheRealDannyyy** [comment]
Has this issue been fixed? From the PR it looks like it's all done.

**Comment by TheRealDannyyy** [comment]
I just tested this out and it's still broken.
@rogerwang This is a P3 issue, would you mind fixing it?

My steps to reproduce (sample: [package.zip](https://github.com/nwjs/nw.js/files/4858316/package.zip)):
1. Run black app (includes "--user-data-dir=test")
2. Close running app
3. Notice "test" folder being created as expected
4. Notice "UserdirTEST" folder also being created, despite using custom path 
[C:\Users\YOURNAMEHERE\AppData\Local\UserdirTEST]

**Comment by mkilburn60** [comment]
@TheRealDannyyy yes its working really well ;). Be careful to use single quotes to handle paths with spaces. Here is an example with single quotes in two places.  "chromium-args": "--user-data-dir='%APPDATA%'\\TheProduct\\'User Data'"

**Comment by TheRealDannyyy** [comment]
@mkilburn60 Based my test case it doesn't. It seems like crashpad is causing a userdir to be created inside the appdata folder.

**Comment by mkilburn60** [comment]
@TheRealDannyyy note you two forward slash for each of the path separators but this page removes one when it renders.

**Comment by mkilburn60** [comment]
@TheRealDannyyy thats true - crash dumps still go to the default.

**Comment by mkilburn60** [comment]
@TheRealDannyyy I'm guessing the same fix would need to be applied to --crash-dumps-dir

**Comment by mkilburn60** [comment]
@TheRealDannyyy the expansion is applied to the whole command line so try setting --crash-dumps-dir to %APPDATA%

**Comment by webstorybegin** [comment]
Hi guys, is the problem still not solved? 
when installing our application, a folder like nw_817_813871837 (numbers may vary) is created in c:/Program Files, it is empty. If you turn off (close) the application this folder disappears. If you uninstall and reinstall the folder is not created. If you install another version of the application, the folder appears again. Sometimes it is not deleted after closing.
Windows defender reacts to all this. Very much need help, in our application windows defender plays a big role

**Comment by iopenet** [comment]
**A portable app shouldn't leave any junk behind!**

Accrding to the docs : 

[Windows](https://www.chromium.org/developers/crash-reports/#windows)
Windows uses crashpad by default. It's compiled in and enabled by default.

[Linux](https://www.chromium.org/developers/crash-reports/#linux)
Breakpad is compiled in, but is not enabled by default. See IsCrashReportingEnabled() in chrome_browser_main_linux.cc for details on when it is enabled.

[Mac](https://www.chromium.org/developers/crash-reports/#mac)
Mac uses Crashpad as its crash reporter instead of Breakpad. Crashpad is always compiled in and enabled, though reports are only stored locally and not uploaded by default.

++
**Chrome on Ubuntu Fails to Respect --crash-dumps-dir and --disable-crash-reporter Flags**  
https://github.com/GoogleChromeLabs/chrome-for-testing/issues/114#issue-2157471548

> Description:
> We are experiencing an issue with Google Chrome on an Ubuntu system, where Chrome attempts to write or create directories under /var/lib/${USER_NAME}, leading to an EACCES (Permission Denied) error. This behavior occurs despite running Chrome under a specific user account with restricted permissions.
> 
> Steps to Reproduce:
> Log in or switch to a non-root user account on an Ubuntu system.
> Launch Google Chrome from the terminal using the user account, including either the --crash-dumps-dir flag to specify a custom directory for crash reports or the --disable-crash-reporter flag to disable the crash reporter feature.
> Observe that Chrome attempts to write or create directories under /var/lib/${USER_NAME}, resulting in an EACCES error and browser crashing

**Comment by stereokai** [comment]
Is this solved yet?
